mgplvm/rdist/rGP.py

```diff
@@ -202,15 +196,18 @@ def concentration_parameters(self):
         return [self._scale, self._ell]
 
     def lat_prms(self, Y=None, batch_idxs=None, sample_idxs=None):
-        mu, K_half_S = self.prms
+        mu, K_half = self.prms
         if batch_idxs is not None:
             mu = mu[:, batch_idxs, :]
-            K_half_S = K_half_S[..., :, batch_idxs][..., batch_idxs, :]
+            if self.use_fast_toeplitz:
+                K_half = K_half[..., batch_idxs]
+            else:
+                K_half = K_half[..., :, batch_idxs][..., batch_idxs, :]
         if sample_idxs is not None:
             mu = mu[sample_idxs, ...]
-            K_half_S = K_half_S[sample_idxs, ...]
+            K_half = K_half[sample_idxs, ...]
```

Comment:
Would this break if sample_idxs break grid structure though?

Reply:
(and K_half is toeplitz structured)

Reply:
(if not maybe add a test for this anyhow)

Reply:
the 'samples' dimension is just a list a list of posteriors but trial, so what we're doing here is sub-selecting the covariance matrices for a subset of trials. Think this should be fine irrespective of the structure of K_half?

Reply:
ah ok, yeah that should be fine -- I thought samples sub-selected time points, which would have broken the symmetry.

Is K_half always the same for every sample_idx? or is this built to allow for different time points for different trials?

Reply:
it's built to allow different time points for different trials, but we should probably built in functionality to not do redundant calculations when they are in fact identical

Reply:
That seems like a different PR, filed an issue #57

Reply:
agreed, sounds good!